doc/contributing.rst

```diff
@@ -1,221 +1,760 @@
-coding is done saves everyone time and effort!
+1. synchronize your local copy with ``upstream``
 
-Installing developer version
-----------------------------
+2. make changes to your local copy
 
-The standard installation procedure can be found on
-:ref:`install_python_and_mne_python` but if you want to install a
-developer version directly from the source code in order to contribute,
-you can follow those steps:
+3. `push`_ your changes to ``origin`` (your remote fork of the upstream)
 
-  .. code-block:: console
+4. submit a `pull request`_ from your fork into ``upstream``
 
-     $ git clone git@github.com:mne-tools/mne-python.git
-     $ cd mne-python
-     $ conda env create -f environment.yml -n mne-dev
-     $ conda activate mne-dev
-     $ pip install -e .  # or python setup.py develop
+The sections :ref:`basic-git` and :ref:`github-workflow` (below) describe this
+process in more detail.
 
-Note: -n flag provides the name for the new (development) environment, and
-overrides any name specified in the .yml file.
-To check the installation, you can enter the following commands:
 
-  .. code-block:: console
+Setting up your local environment for MNE-Python development
+============================================================
 
-     $ python -c "import mne; mne.sys_info()"
-     $ python -c "import mne; mne.datasets.testing.data_path(force_update=True)"
+Configuring git
+^^^^^^^^^^^^^^^
 
-Code guidelines
----------------
+To get set up for contributing, make sure you have git installed on your local
+computer:
 
-* Standard Python style guidelines set by pep8_ and pyflakes_ are followed
-  with very few exceptions. We recommend using an editor that calls out
-  style violations automatcally, such as Spyder_. From the MNE code root, you
-  can check for violations using flake8_ with:
+- On Linux, the command ``sudo apt install git`` is usually sufficient; see the
+  `official Linux instructions`_ for more options.
 
-  .. code-block:: console
+- On MacOS, download `the .dmg installer`_; Atlassian also provides `more
+  detailed instructions and alternatives`_ such as using MacPorts or Homebrew.
 
-     $ make flake
+- On Windows, we recommend `git Bash`_ rather than the `official Windows
+  version of git`_, because git Bash provides its own shell that includes many
+  Linux-equivalent command line programs that are useful for development.
+  Windows 10 also offers the `Windows subsystem for Linux`_ that offers similar
+  functionality to git Bash, but has not been widely tested by MNE-Python
+  developers yet.
 
-* Use `numpy style`_ for docstrings. Follow existing examples for simplest
-  guidance.
+.. note::
 
-* New functionality must be covered by tests. For example, a
-  :class:`mne.Evoked` method in ``mne/evoked.py`` should have a corresponding
-  test in ``mne/tests/test_evoked.py``.
+    `GitHub desktop`_ is a GUI alternative to command line git that some users
+    appreciate; it is available for Windows and MacOS.
```

Comment:
Too many `.. note::` sections messes with the flow and readability:

![Screenshot from 2019-05-02 10-23-34](https://user-images.githubusercontent.com/2365790/57082295-72632200-6cc4-11e9-8ccb-0859556ba651.png)

If any of these can be shortened and added as normal text, it can help.

The best solution might be to find a CSS way to make these collapsable sections that people can opt-in to looking at. Like a `Details` sort of click-to-expand thing, but where you can set the title to something relevant. WDYT? It might be worth looking into now, since it seems like we could do this many places.